### PR TITLE
feat(tag): introduce tag

### LINKED ISSUE #178: Component: Tag

### PR DESCRIPTION
**Pull request by emyarod**
Closes #178

### CODE REVIEW

File: src/components/tag/filter-tag.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, customElement, LitElement } from 'lit-element';
+import classnames from 'classnames';
+import settings from 'carbon-components/es/globals/js/settings';
+import Close16 from '@carbon/icons/lib/close/16';
+import FocusMixin from '../../globals/mixins/focus';
+import TAG_TYPE from './types';
```
Comment (asudoh) [inline src/components/tag/filter-tag.ts]: Great to have this types re-exported from this file.

File: src/components/tag/filter-tag.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * Filter tag.
+ */
+@customElement(`${prefix}-filter-tag`)
+export default class BXFilterTag extends FocusMixin(LitElement) {
+  /**
+   * Text to show on filter tag "clear" buttons. Corresponds to the attribute with the same name
+   */
+  @property({ type: String, reflect: true })
+  title = '';
```
Comment (asudoh) [inline src/components/tag/filter-tag.ts]:
```suggestion
  title!: string;
```

Comment (asudoh) [reply src/components/tag/filter-tag.ts]: Another possible approach is `title = 'Clear filter'`.

File: src/components/tag/filter-tag.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  disabled = false;
+
+  /**
+   * Specify the type of the <Tag>
+   */
+  @property({ reflect: true })
+  type = TAG_TYPE.RED;
+
+  render() {
+    const { disabled, title, type } = this;
```
Comment (asudoh) [inline src/components/tag/filter-tag.ts]:
```suggestion
    const { disabled, title = 'Clear filter', type } = this;
```

File: src/components/tag/filter-tag.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    const classes = classnames(`${prefix}--tag`, `${prefix}--tag--${type}`, `${prefix}--tag--filter`, {
+      [`${prefix}--tag--disabled`]: disabled,
+    });
+    return html`
+      <span class="${classes}" title="${title || 'Clear filter'}" tabindex="0">
```
Comment (asudoh) [inline src/components/tag/filter-tag.ts]:
```suggestion
      <span class="${classes}" title="${title}" tabindex="0">
```

File: src/components/tag/filter-tag.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        <slot></slot>
+        ${Close16({ 'aria-label': title || 'Clear filter' })}
```
Comment (asudoh) [inline src/components/tag/filter-tag.ts]:
```suggestion
        ${Close16({ 'aria-label': title })}
```

File: src/components/tag/filter-tag.ts (hunk 3 above)
Comment (emyarod) [reply src/components/tag/filter-tag.ts]: this won't be needed after [adding this change](https://github.com/carbon-design-system/carbon-custom-elements/pull/179#discussion_r349996941) right?

File: src/components/tag/filter-tag.ts (hunk 1 above)
Comment (emyarod) [reply src/components/tag/filter-tag.ts]: exported `TAG_TYPE` from `tag.ts` and `filter-tag.ts` and removed some `./tag` imports, let me know if the imports and exports look correct now

File: src/components/tag/filter-tag.ts (hunk 3 above)
Comment (asudoh) [reply src/components/tag/filter-tag.ts]: That's right!

**Pull request merged**